### PR TITLE
Billy/1054 my stake page

### LINKED ISSUE #1054: implement "My Stake" page
New design 

screen shot 2018-09-03 at 4 07 44 pm

https://zpl.io/aBNdBxK

### PR DESCRIPTION
**Pull request by okwme**
Closes #1054
with `.red`, `.green` and `.orange` classes added to some of the values it looks like this. I added back the checkbox since as of now there's no other way to get to those pages. The tab is disabled for "My Stake" because that task is now part of https://github.com/cosmos/voyager/issues/1281, https://github.com/cosmos/voyager/issues/1278, https://github.com/cosmos/voyager/issues/1280.

screenshot 2018-09-04 12 38 32

### CODE REVIEW

**Comment by okwme** [comment]
@faboweb and i discussed and it's ok to approve this PR even tho it doesn't pass codecoverage because the only part of `LiDelegate` not being tested is a part which should actually be removed as per this issue: https://github.com/cosmos/voyager/issues/1276

File: app/src/renderer/scripts/common.js [hunk 1]
```diff
@@ -1,3 +1,16 @@
 module.exports.sleep = function(ms) {
   return new Promise(resolve => setTimeout(resolve, ms))
 }
+
+module.exports.shortAddress = function(address, length = 4) {
```
Comment (faboweb) [inline app/src/renderer/scripts/common.js]: maybe add a comment on what this does

File: app/src/renderer/components/staking/PanelSort.vue [hunk 2]
```diff
@@ -53,36 +54,43 @@ export default {
...
+      // margin-left auto
   .label
-    color var(--link)
-    white-space nowrap
-    text-overflow ellipsis
-    overflow hidden
+    display inline-block
+    margin auto
+    &:first-of-type .label-text:after
+      // content ''
```
Comment (faboweb) [inline app/src/renderer/components/staking/PanelSort.vue]: do we need to keep those comments?

File: tasks/runner.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ function startRendererServer() {
 }
 
 module.exports = async function(networkPath) {
+  console.log(networkPath)
```
Comment (faboweb) [inline tasks/runner.js]: do we need that?

File: tasks/testnet.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ async function main() {
   const network = process.argv[2] || config.default_network
 
   // run Voyager in a development environment
-  runDev(`./app/networks/${network}/`)
+  runDev(`./builds/testnets/${network}/`)
```
Comment (faboweb) [inline tasks/testnet.js]: is this hiding the problem?

File: app/src/renderer/components/staking/LiDelegate.vue [hunk 5]
```diff
@@ -116,23 +137,68 @@ export default {
 
 .li-delegate__values
   display flex
-  height 3rem
+  height 5rem
+  padding 12px 1em
+  background-color var(--app-nav)
+
+  & > .li-delegate__value:not(:first-of-type) span
```
Comment (jbibla) [inline app/src/renderer/components/staking/LiDelegate.vue]: highly impressive css 👏 🥇

File: app/src/renderer/components/staking/LiDelegate.vue [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
     span {{ yourVotes }}
-  .li-delegate__value.status
-    span {{ delegateType }}
+  .li-delegate__value.your-rewards
+    span {{ yourRewards }}
+  .li-delegate__break: span
+  .li-delegate__value.percent_of_vote
+    span {{ delegate.percent_of_vote }}
+  .li-delegate__value.uptime
+    span.NOT_green {{ uptime }}
```
Comment (jbibla) [inline app/src/renderer/components/staking/LiDelegate.vue]: what is this `NOT` about?

File: app/src/renderer/components/staking/PageStaking.vue [hunk 7]
```diff
@@ -7,11 +7,15 @@ tm-page(title='Staking')
...
+    .tab(v-for="(tab, i) in tabs",
+      :key="'tab-' + i",
+      :class="{'tab-selected': i === tabIndex}",
+      @click="tabIndex = 1") {{tab}}
   .delegates-container
-    tm-data-loading(v-if="delegates.loading && delegates.delegates.length === 0")
-    tm-data-empty(v-else-if="delegates.delegates.length === 0")
-    data-empty-search(v-else-if="sortedFilteredEnrichedDelegates.length === 0")
+    tm-data-loading(v-if="delegates.loading && sortedFilteredEnrichedDelegates.length === 0")
+    tm-data-empty(v-else-if="!delegates.loading && delegates.delegates.length === 0")
```
Comment (jbibla) [inline app/src/renderer/components/staking/PageStaking.vue]: should `delegates.delegates.length` be `sortedFilteredEnrichedDelegates.length`?

File: app/src/renderer/scripts/common.js (hunk 1 above)
Comment (okwme) [reply app/src/renderer/scripts/common.js]: 👍

File: app/src/renderer/components/staking/PanelSort.vue (hunk 2 above)
Comment (okwme) [reply app/src/renderer/components/staking/PanelSort.vue]: 👎

File: tasks/runner.js (hunk 3 above)
Comment (okwme) [reply tasks/runner.js]: 👎

File: tasks/testnet.js (hunk 4 above)
Comment (okwme) [reply tasks/testnet.js]: 🙃

File: app/src/renderer/components/staking/LiDelegate.vue (hunk 5 above)
Comment (okwme) [reply app/src/renderer/components/staking/LiDelegate.vue]: be wary of `:first-of-type` it only refers to native elements, not classes!!!

File: app/src/renderer/components/staking/LiDelegate.vue (hunk 6 above)
Comment (NodeGuy) [reply app/src/renderer/components/staking/LiDelegate.vue]: I didn't quite get that.  Would you please explain it again in German?

Comment (faboweb) [reply app/src/renderer/components/staking/LiDelegate.vue]: I guess those are placeholders. I replace them with comments.

File: app/src/renderer/components/staking/PageStaking.vue (hunk 7 above)
Comment (jbibla) [reply app/src/renderer/components/staking/PageStaking.vue]: smells like a refactor (one day).

`delegates.delegates.delegates.delegates.delegates.goose`

🦆 🦆 🦆 🦆 🦆

Comment (faboweb) [reply app/src/renderer/components/staking/PageStaking.vue]: didn't we have the discussion before? ;)

**Pull request merged**